intercom-java/src/main/java/io/intercom/api/Visitor.java

```diff
@@ -0,0 +1,770 @@
+package io.intercom.api;
+
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Maps;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@SuppressWarnings("UnusedDeclaration")
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+public class Visitor extends TypedData {
+
+    private static final Map<String, String> SENTINEL = Maps.newHashMap();
+
+    public static Visitor findByID(String id)
+            throws AuthorizationException, ClientException, ServerException, InvalidException, RateLimitException {
+        return new HttpClient(visitorURI(id)).get(Visitor.class);
+    }
+
+    public static Visitor findByUserID(String userID)
+            throws AuthorizationException, ClientException, ServerException, InvalidException, RateLimitException {
+        final Map<String, String> params = Maps.newHashMap();
+        params.put("user_id", userID);
+        return DataResource.find(params, "visitors", Visitor.class);
+    }
+
+    public static Visitor update(Visitor v)
+            throws AuthorizationException, ClientException, ServerException, InvalidException, RateLimitException {
+        return DataResource.updatePut(VisitorUpdate.buildFrom(v), visitorURI(""), Visitor.class);
+    }
+
+    public static Visitor delete(Visitor v)
+            throws AuthorizationException, ClientException, ServerException, InvalidException, RateLimitException {
+        return DataResource.delete(v.getID(), "visitors", Visitor.class);
+    }
+
+    public static Visitor delete(String id)
+            throws AuthorizationException, ClientException, ServerException, InvalidException, RateLimitException {
+        return DataResource.delete(id, "visitors", Visitor.class);
+    }
+
+    public static Contact convertToContact(Visitor v)
+            throws AuthorizationException, ClientException, ServerException, InvalidException, RateLimitException {
+        return DataResource.post(VisitorConvertBuilder.buildConvertToLead(v), convertURI(), Contact.class);
+    }
+
+    public static User convertToUser(Visitor v, User u)
```

Comment:
Since we pass a `User`, I believe this will [merge](https://developers.intercom.com/intercom-api-reference/reference#convert-a-visitor-to-a-user) and not convert. That's a confusing aspect of our SKDs (overall, not only this one) vs the API, so is that worth renaming the function to `mergeWithContact`? /cc @choran

Reply:
You can actually convert a visitor to a user if the user identifier doesn't exist.

> If the User does not exist, the Visitor will be converted to a User, with the User identifiers replacing it's Visitor identifiers.


```
User user = new User();
user.setID("New User ID that hasn't been used");
Visitor.convertToUser(visitor,user);
```

![image](https://user-images.githubusercontent.com/892961/46525646-c472cf00-c8be-11e8-83ac-0838495d6f49.png)

I believe your scenario is the other part mentioned in the docs
> If the User exists, then the Visitor will be merged into it, the Visitor deleted and the User returned. 

But yeah it is a bit confusing not knowing how the merge will work until you get the responding record and compare the Intercom ID with the original visitor record

Reply:
Ah good point, I was not sure how that would behave in the first case w/ the SDK. 👍 in that case!